array-methods.js

```diff
@@ -83,38 +104,99 @@ var sumOfInterests = null;
     round this number to the nearest 10th of a cent before moving on.
   )
  */
-var stateSums = null;
+
+var stateSums = dataset.bankBalances.reduce( (obj, accounts) => {
+  if(!obj.hasOwnProperty(accounts.state)) {
+    obj[accounts.state] = 0;
+  }
+
+  obj[accounts.state] = parseFloat((parseFloat(obj[accounts.state]) + parseFloat(accounts.amount)).toFixed(2));
+  return obj;
+ }, {});
+
+var stateSumsArray = Object.keys(stateSums)
+.map ( stateKey => {
+  return {
+      state : stateKey,
+      amount : stateSums[stateKey]
+    };
+});
 
 /*
-  from each of the following states:
+  excluding each of the following states:
+
     Wisconsin
     Illinois
     Wyoming
     Ohio
     Georgia
     Delaware
+
   take each `amount` and add 18.9% interest to it
-  only sum values greater than 50,000 and save it to `sumOfInterests`
+  only sum values if interest is greater than 50,000 and save it to `sumOfHighInterests`
 
   note: During your summation (
     if at any point durig your calculation where the number looks like `2486552.9779399997`
     round this number to the nearest 10th of a cent before moving on.
   )
  */
-var sumOfHighInterests = null;
+
+var partialStateSums = dataset.bankBalances.filter( ( obj ) => {
+    return obj.state !== 'WI' && obj.state !== 'IL' && obj.state !== 'WY' && obj.state !== 'OH' && obj.state !== 'GA' && obj.state !== 'DE';
+})
+.reduce( (obj, {state, amount}) => {
+  if(!obj.hasOwnProperty(state)) {
+    obj[state] = 0;
+  }
+  obj[state] =  Math.round(obj[state]) + Math.round(amount) * 0.189;;
+  return obj;
+ }, {});
+
+var sumOfHighInterests = Object.keys(partialStateSums)
+.map ( stateKey => {
+  return {
+      amount : (partialStateSums[stateKey] * 18.9) / 100,
+      state : stateKey
+    };
+})
+.filter ( sums  => {
+  return sums.amount > 50000;
+})
+.reduce ( ( sum, accounts) => {
+  return parseFloat(accounts.amount + sum);
```

Comment:
I don't think you need to use parseFloat here since the values are still numbers at this point.